ecommerce/extensions/payment/tests/processors/test_paystack.py

```diff
@@ -0,0 +1,174 @@
+""" Unit tests of Paystack payment processor implementation. """
+import responses
+from mock import patch
+from oscar.apps.payment.exceptions import GatewayError
+
+from ecommerce.extensions.payment.exceptions import RefundError
+from ecommerce.extensions.payment.processors.paystack import Paystack
+from ecommerce.extensions.payment.tests.processors.mixins import PaymentProcessorTestCaseMixin
+from ecommerce.extensions.test.paystack_utils import (
+    get_error_response,
+    get_initliaze_payment_response,
+    get_refund_create_response,
+    get_refund_fetch_response,
+    get_transaction_response
+)
+from ecommerce.tests.testcases import TestCase
+
+
+class PaystackTests(PaymentProcessorTestCaseMixin, TestCase):
+    """
+    Tests for the Paystack payment processor.
+    """
+    processor_class = Paystack
+    processor_name = 'paystack'
+
+    @classmethod
+    def setUpClass(cls):
+        super(PaystackTests, cls).setUpClass()
+
+    def setUp(self):
+        super(PaystackTests, self).setUp()
+        self.base_url = self.processor.configuration['base_url']
+        self.reference_number = 'abcdefghi'
+        self.transaction_id = '111111111111'
+        self.refund_id = '22222222222'
+
+    @responses.activate
+    def test_get_transaction_parameters(self):
+        """
+        Verifies that the processor return appropriate payment url.
+        """
+        token = 'fake_token'
+        expected_payment_url = '{}/{}'.format(self.base_url, token)
+
+        response_data = get_initliaze_payment_response(expected_payment_url, token, self.reference_number)
+        url = '{}/transaction/initialize'.format(self.base_url)
+        responses.add(responses.POST, url, json=response_data, status=200)
+
+        with patch('ecommerce.extensions.payment.processors.paystack.logger') as mock_logger:
+            actual_payment_url = self.processor.get_transaction_parameters(self.basket).get('payment_page_url')
+            mock_logger.info.assert_called_once_with(
+                "Successfully got hosted Paystack payment page for basket: %d.", self.basket.id)
+            self.assertEqual(actual_payment_url, expected_payment_url)
+
+    @responses.activate
+    def test_get_transaction_parameters_error(self):
+        """
+        Verifies that the processor raise GatewayError if it is unable to get payment URL from Paystack.
+        """
+        response_data = get_error_response()
+        url = '{}/transaction/initialize'.format(self.base_url)
```

Comment:
Can we make these URLs using `reverse` ?

Reply:
nope, as these are Paystack API URLs.